### ISSUE TITLE
ignoreFiles config value is ignored

### ISSUE THREAD

**Issue by paulrobertlloyd** [AtomLinter/linter-stylelint#84]
Given the config:

```json
{
    "extends": "/Volumes/Macintosh/Users/paulrobertlloyd/.atom/packages/linter-stylelint/node_modules/stylelint-config-standard",
    "ignoreFiles": [
        "source/_assets/stylesheets/vendor/*.scss"
    ],
    "rules": {
        "at-rule-empty-line-before": [ "always", {
            "except": ["blockless-group", "first-nested"]
        } ],
        "declaration-colon-newline-after": "always-multi-line",
        "function-url-quotes": "none",
        "indentation": 4
    }
}
```

and folders/files such:

```
source
└─ _assets
   └─ stylesheets
      └─ vendor
         └─ _rouge.scss
```

When editing _rouge.scss, linting errors are still reported. Running stylelint directly from the CLI however, the `ignoreFiles` value is respected.

**Comment by Arcanemagus** [comment]
This is due to us using the Node API's `lint()` call on the raw text (to support linting on the fly), instead of a file.

@davidtheclark is there anyway to specify the filename of a block of code so `stylelint` will check it against this, or should I file an issue about that over there? :stuck_out_tongue:

**Comment by davidtheclark** [comment]
@Arcanemagus I hadn't run into this need before. May as well file an issue with stylelint. But I can't think of any use-case for this other than text editors figuring out whether to ignore the file??? Can anyone else? 

I guess there could be a `from` option in the node API for specifying a source file name that should be attributed to the direct `code` input string, and if it's in the `ignoreFiles` glob it will be ignored.

Isn't another way to handle this to expect the text editor plugins to read the `ignoreFiles` glob and determine whether to run stylelint or not?

Or I guess another possibility is that instead of calling stylelint with `code` you actually provide it with the filename. It then finds the file, loads the config, checks the ignoreFiles, etc. Or would that slow things down too much? I don't know: wouldn't it only add one `fs` call (to get the file)?

**Comment by Arcanemagus** [comment]
The problem with a file is linting on the fly, we would need to use a temporary file (the helper library makes this simple), I just don't know how involved other files are in the checking process. If linting a file _only_ involves that file, it would be relatively simple to have the helper library copy to a temporary directory and then run on that.

**Comment by davidtheclark** [comment]
@Arcanemagus Interesting. Well, stylelint wouldn't need anything other than the file to run. We could look into that option at some point.

**Comment by Arcanemagus** [comment]
As an FYI, several linters that provide a Node API, and are able to ignore files based on matching a pattern, support specifying the file name for code/stdin... but there are also a few that don't where it has had to be worked around in the provider here.

**Comment by davidtheclark** [comment]
@Arcanemagus: Could you point me to some linters that do? I'd like to see what the conventions are.

**Comment by Arcanemagus** [comment]
@1000ch Do you want to work on moving to the temporary file method?

@davidtheclark `eslint` is the first one that pops into mind: http://eslint.org/docs/developer-guide/nodejs-api

**Comment by davidtheclark** [comment]
@Arcanemagus: This is a pretty easy addition to stylelint. I will get a PR up before breakfast.

**Comment by 1000ch** [comment]
@paulrobertlloyd This may be resolved in `v1.13.0`. Thanks @davidtheclark :+1:

**Comment by paulrobertlloyd** [comment]
Yup, it works. Fantastic, thank-you!
